Add idpDB and HDF5 sniffers and fix MzSQlite sniffer

chambm opened the pull request:
Move idpDB and MzSQlite formats from proteomics.py to binary.py, because the SQLite sniffer lives there, and more specialized formats must be registered for sniffing before the generic format

Add proper idpDB sniffing logic (akin to MzSQlite) and test case
Add H5 sniffing based on magic number and test case

bgruening committed:
add BAM iobio visualisation as Interactive Environment

jmchilton committed:
Refactor test dataset populator for reuse on jobs without outputs.

jmchilton committed:
Add a quick shortcut for starting Galaxy with development tools.

Start Galaxy using

    GALAXY_RUN_WITH_TEST_TOOLS=1 run.sh

Enables beta tool formats (e.g. YAML, and CWL downstream), beta workflow modules, and the suite of Galaxy test tools used by testing framework.

natefoo committed:
Merge branch 'dev' into pulsar-client-update

carlfeberhard committed:
Client logging: adapt metrics logger for optional debug logging

carlfeberhard committed:
Client logging: allow focusing on one or more namespaces, add function to more easily namespace loggers

natefoo committed:
Merge linting and tool loader changes from Planemo

natefoo committed:
PEP-8 fix for linters

natefoo committed:
Merge Planemo and Galaxy tools.deps changes (and a PEP-8 fix)

natefoo committed:
Correct a mistakenly reformatted comment. Inline isn't possible because
hanging an inline violates E113 (although PEP-8 does not explicitly say
it's not allowed, but it does say "An inline comment is a comment on the
same line as a statement"), so convert the comments to block.

jj-umn committed:
Add uniprotxml datatype

jj-umn committed:
Add UniProtXML datatype to proteomics.py

natefoo committed:
Python 3-ify the object store (from Pulsar, for Planemo)

natefoo committed:
Add 2 missing dependencies of dependencies

carlfeberhard committed:
Client logging: namespace current loggables

martenson committed:
Merge pull request #916 from dannon/pref_bump

Change user preference datatype to text

hexylena committed:
Log TS installation commands with repository name

Fixes #436

guerler committed:
Insert promises

Revision

Revise deferred class to use promises

Update charts

Fix comment

Use resolve and reject properly

Fix comments

Fix comments

Fix comments

Fix comment

Improve naming/comments

Remove unnecessary queue handler

Fix naming

hexylena committed:
Replace remaining print statements with log.*

guerler committed:
Undo 965 for dev

guerler committed:
Repack scripts

jmchilton committed:
Remove old XSD linting code.

jmchilton committed:
Restore changes to tool shed package resolver.

carlfeberhard committed:
Client logging: remove 'namespaced', allow 'log' to be added in the standard way (now that browsers are compat with this), simplify interface in the Galaxy object

carlfeberhard committed:
Merge remote-tracking branch 'upstream/dev' into logging: attempting to fix qunit failures

hexylena committed:
Fix flake8 errors, @nsoranzo

nsoranzo committed:
Apply stricter flake8 rules for the directories shared with Pulsar

hexylena committed:
Fix review comments

carlfeberhard committed:
Client logging: move logger assignment into Galaxy object, both Galaxy.user and all LoggableMixins will now respond to debugging, properly localize error messages

jxtx committed:
Labels for individual tools in tool panel.

Allow one or more labels to be attached to a tool in the tool panel.
These will be rendered to the right of the tool's link using the
boostrap label style. Styles for specific labels can be defined in CSS,
specific colored labels for "new" and "beta" are defined.

jxtx committed:
Make toolbox searchable by tool labels.

martenson committed:
Merge pull request #968 from guerler/fix_require_login_003

Show masthead if login is required, add login page to front end

jxtx committed:
Updates to default welcome page.

Reference CONTRIBUTORS.md, update institution references, update to
HTML5, use some standard classes for formatting, include links for
configuring and installling tools.

jxtx committed:
Change a.btn style, fix welcome link targets.

"a" elements styled with .btn will have no underline by default.

jxtx committed:
Fix link target in welcome

martenson committed:
Merge pull request #1013 from jxtx/fix-default-welcome-references

Updates to default welcome page.

natefoo committed:
Merge pull request #1 from jmchilton/pulsar-planemo-lib-update

Remove old XSD linting code.

natefoo committed:
Merge pull request #990 from nsoranzo/post_eggs_smash

Remove remaining references to galaxy.eggs . flake8 some tools.

natefoo committed:
Merge pull request #1008 from erasche/installation_logging

Log TS installation commands with repository name

jmchilton committed:
Merge pull request #1005 from natefoo/pulsar-planemo-lib-update

Merge changes to common modules shared between Galaxy, Planemo, and Pulsar.

natefoo committed:
Add importlib and ordereddict for Python 2.6.

They won't be installed by conditional since they will always be
installed as dependencies of kombu in requirements.txt. This should be
fixed at some point.

natefoo committed:
XPath support is not the same on all Python 2.6es

jj-umn committed:
Add UniProtXML sniffer tag

dannon committed:
Downgrade WebError.  This fixes the interactive debugger (for now -- to be replaced w/ werkzeug eventually)

dannon committed:
Remove weberror/pygments from unconditional dependencies list

guerler committed:
Fix display call

Fix config

Fix naming

guerler committed:
Merge branch 'dev' of https://github.com/galaxyproject/galaxy into revise_promise_000

guerler committed:
Keep select field open for multi select

dannon committed:
Merge pull request #1009 from erasche/print-logging

Replace remaining print statements with log.*

dannon committed:
Merge pull request #1010 from nsoranzo/flake8_fixes

Apply stricter flake8 rules for the directories shared with Pulsar

dannon committed:
Manual merge and rebuild due to conflict.

dannon committed:
Merge pull request #1011 from carlfeberhard/logging

Client side logging

guerler committed:
Use remove instead of empty if iframe is shown

dannon committed:
Merge pull request #1018 from guerler/revise_promise_000

Revise deferred queue, use jQuery promises

guerler committed:
Patch select2 to avoid resetting highlighted location

Keep search term

dannon committed:
Merge pull request #1014 from jmchilton/cwl_util

Two Small Tweaks from CWL Branch

guerler committed:
Allow resizing of regular multi-select boxes

Remove event handler

Add mouse leave event handler

Improve visibility and accessibility

Fix tag name

Style compromise

Repack charts

guerler committed:
Fix upload button style

More fixes

guerler committed:
Fix style of labels in grids

Fix styles

Pack styles

yhoogstrate committed:
Fixes broken env-var declarations for tools with weird chars in their name

dannon committed:
Merge pull request #1027 from guerler/fix_grids_000

Fix style of menu button labels in grids

dannon committed:
Rebuild client styles, merge.

yhoogstrate committed:
Replacement of dots

hexylena committed:
Merge pull request #1028 from yhoogstrate/step_handler_fix

Fixes broken env-var declarations for tools with weird chars in their…

guerler committed:
Merge branch 'dev' of https://github.com/galaxyproject/galaxy into revise_multiselect_001

dannon committed:
Merge pull request #1025 from guerler/revise_multiselect_001

Allow resizing of regular multi-select boxes

guerler committed:
Merge branch 'dev' of https://github.com/galaxyproject/galaxy into revise_multiselect_000

guerler committed:
Pack scripts

scholtalbers committed:
wig_to_bigwig: Add error message for wrongly formatted data

nsoranzo committed:
Merge pull request #1033 from scholtalbers/wig_to_bigwig_error_msg

wig_to_bigwig: Add error message for wrongly formatted data

jmchilton committed:
Introduce "Format 2" workflow definitions.

This is a slight tightening up of the ad-hoc, experimental workflow YAML definition used by the test framework (and by Kyle, but lets just admit Kyle is part of Galaxy's test framework).

This format is still defined entirely client side by transcoding the YAML or python object description into real (or format 1) Galaxy workflows - so these cannot realisticaly be declared part of Galaxy's public interface and can remain experimental and subject to change.

In addition to refactoring the code implementing these workflows for upstream modifications to explore new features, the format itself has been made slightly more stringent in two ways:

 - 'steps' must now be explicit (previously converted auto-convert list to dict).
 - Ensure the workflow declared a 'class' is defined and the class is 'GalaxyWorkflow'.

This will help with discovery of workflow objects with downstream tooling (planemo for workflows?) and brings the Galaxy definition slightly more inline with the CWL definition for workflows (very slightly).

natefoo committed:
Add stub fetch_eggs.py/check_eggs.py for people whose build/test/run
pipelines may have been running these by hand. Print hopefully helpful
information and return 1. Fixes #1030

dannon committed:
Merge pull request #1006 from natefoo/smash_eggs

Add 2 missing dependencies of dependencies

guerler committed:
Fix log comments

Use emit

Fix namespace

Fix namespace

Remove empty line

Check if logger exists

Fix comment

Redo

More

Fix yet another log

Fix statements

More fixes

Move emitter

carlfeberhard committed:
Merge pull request #1019 from guerler/revise_multiselect_000

Keep select field open for multi select

blankenberg committed:
Merge pull request #1037 from natefoo/smash-eggs-bugfix1

Add stub fetch_eggs.py/check_eggs.py

jmchilton committed:
Merge branch 'release_15.10' into dev

Conflicts:
	client/galaxy/scripts/mvc/grid/grid-view.js
	static/maps/mvc/collection/list-of-pairs-collection-creator.js.map
	static/scripts/mvc/collection/list-of-pairs-collection-creator.js

martenson committed:
Merge pull request #1040 from guerler/revise_logging_000

Fix log statements and link to logger

jmchilton committed:
Lint fix for 15.10 to dev merge.

dannon committed:
Add pygments back, since it's a dependency of a conditional dependency.

dannon committed:
Merge and resolve conflicts

yhoogstrate committed:
Added gitignore rules to ignore custom tool-data

jmchilton committed:
Improved unicode handling stdout/stderr/command_line.

dannon committed:
Merge pull request #1048 from yhoogstrate/tool_data_gitignore

Added gitignore rules to ignore custom tool-data

jmchilton committed:
Unify standard error and output trimming and unicoding.

Should be more complete now (were paths for task and jobs where it wasn't be unicodified) and with less duplication. Moved logic into model layer where it probably belongs anyway.

jmchilton committed:
Fix failing tests for output checker changes.

jmchilton committed:
Fixes for c790fcb84d005025b331b2a9106e7097860a7545.

Thanks to @nsoranzo.

jmchilton committed:
Remove more unneeded stuff from galaxy.jobs now.

Thanks to @nsoranzo.

natefoo committed:
Merge pull request #1017 from dannon/weberror_downgrade

Downgrade WebError.

jmchilton committed:
Restore lost call.

Caught by @nsoranzo.

blankenberg committed:
Merge pull request #1044 from jmchilton/release_15.10_to_dev

Bring 15.10 (and 15.07) fixes into dev.

mvdbeek committed:
Fix import of history datasets into library.

mvdbeek committed:
Add changed static files.

blankenberg committed:
Fix for reloading tools that have non-standard tool_ids/versions.

blankenberg committed:
HTML escape tool name, id, version, on tool reload message.

blankenberg committed:
Also escape tool name, id, version in message when removing tools.

martenson committed:
Merge pull request #1049 from mvdbeek/library_add_from_history

Fix import of history datasets into library.

jmchilton committed:
Fix nit picks unrelated to PR #1052...

... as pointed out by @nsoranzo and @blankenberg.

nsoranzo committed:
Fix typo.

nsoranzo committed:
Extend users API with delete action.

nsoranzo committed:
Fix is_admin value in users API show method to show if the requested user is an admin, not the current user.

nsoranzo committed:
Add check for allow_user_deletion.

einonm committed:
Trivial: Replace unnecessary duplicated var check with 'else if'

bgruening committed:
Merge pull request #1052 from jmchilton/unicode_stream_3

Improved Encoding Handling for Jobs

martenson committed:
Merge pull request #1063 from nsoranzo/api_delete_user

Extend users API with delete action.

jmchilton committed:
Refactor tool parsing, parse id first.

Small refactoring cleans up downstream diff for dynamic tool branch.

nsoranzo committed:
Check len files during download from UCSC.

Do not create file in case of errors, so the cron script will not
overwrite a good file with a bad one.

martenson committed:
add @mr-c

dannon committed:
Merge pull request #1067 from martenson/add-biocrusoe

add biocrusoe to contributors

dannon committed:
Fix lped report output from converter

bgruening committed:
Merge pull request #1069 from dannon/fix_lped_report_output

Fix lped report output from converter

dannon committed:
Fix one more hanging div.

carlfeberhard committed:
Merge pull request #1070 from dannon/fix_lped_report_output

Fix one more hanging div in lped to pbed converter.

dannon committed:
Update both lped pbed pbed lped converters to be in sync with their html.  Thanks @nsoranzo.

martenson committed:
Merge pull request #1073 from einon/einon

Trivial: Replace unnecessary duplicated var check with 'else if'

martenson committed:
extend API interactor to allow http PATCH and DELETE testing

martenson committed:
add more API tests for data libraries

dannon committed:
Merge pull request #1050 from blankenberg/reload-tool-id-fix

Fix for reloading tools that have non-standard tool_ids/versions.

blankenberg committed:
Merge pull request #1072 from dannon/fix_lped_report_output

Update both lped pbed pbed lped converters to be in sync with their h…

einonm committed:
upload.py: Remove unused function parameter

martenson committed:
Merge pull request #1078 from einon/einon

upload.py: Remove unused function parameter

martenson committed:
Merge pull request #942 from davebx/docker

Enable dockerized toolshed tests.

davebx committed:
Fix 500 error when attempting to update installed repository.

davebx committed:
Ensure that invalid parameters don't break things.

martenson committed:
remove whitespace at line ends

martenson committed:
update the image tag

Commit:
Resolve conflicting label CSS class.

martenson committed:
Merge pull request #1086 from jgoecks/trackster-pos-label

Resolve conflicting label CSS class for trackster.

davebx committed:
Use preexisting library to parse query string.

natefoo committed:
Newer versions of WebOb set None in referer, older versions set an empty
string. Handle cases where it is assumed to be a string.

I checked all other trans.request attributes to find other cases where
this might crop up and didn't find any.

martenson committed:
Merge pull request #1082 from davebx/hg_middleware

Fix 500 error when attempting to update installed repository.

bgruening committed:
Fix API installation of tool_dependencies

This should fix an installation issue with tool_dependencies via the API.
Tool dependency repositories to not have any tools specified and with this no metadata associated.

It was broken in https://github.com/galaxyproject/galaxy/commit/bb6a784f13ff3f91762244d4488437ffef9e5acf
So I think this should be part of 15.10.

Thanks to Greg for fixing this one.

bgruening committed:
Don't fail if a repository is already installed

If an repository is already installed `installed_tool_shed_repositories` is None and this does not work with map.

jmchilton committed:
Add tool version to workflow extract.

jmchilton committed:
Add app property to workflow contents manager.

Most other managers already have this.

jmchilton committed:
Improve the API attributes on parameters page.

Make clear what API ID is and add display of API value for job.

martenson committed:
Merge pull request #1098 from jmchilton/show_param

Improve the API attributes on parameters page.

martenson committed:
Merge pull request #1094 from bgruening/fix_api_ts_installation

Fix api TS installation

martenson committed:
Merge pull request #1095 from bgruening/ts_api_fail_if_installed

Don't fail if a repository is already installed

nsoranzo committed:
Update run_tests.sh help for #844.

nsoranzo committed:
Merge pull request #1020 from dannon/issue_procedure

[PROCEDURES] Issue procedure linkage

martenson committed:
Merge pull request #1101 from nsoranzo/run_tests_sh

Update run_tests.sh help for #844.

dannon committed:
Rebuild styles, fixes history panel green being wrong.

guerler committed:
Merge pull request #1102 from dannon/style_rebuild

Rebuild styles, fixes history panel green being wrong.

guerler committed:
Show welcome page with required login, remove iframes

Reinsert iframe for login

Use iframes

lparsons committed:
Better error when tool shed repo dir missing

Check for a tool shed repository directory and throw exception if it's
missing. This avoids a cryptic NoneType error when searching for
tool_dependencies.xml file and directory was not found.

dannon committed:
Merge pull request #1105 from guerler/revise_require_login_003

Show welcome page with required login

dannon committed:
Merge pull request #1107 from lparsons/tool_shed_repo_dir_check

Better error when tool shed repo dir missing

yhoogstrate committed:
CRAM format

yhoogstrate committed:
flake8

blankenberg committed:
Merge pull request #1083 from martenson/update-testing-docker-image

Update testing docker image

jmchilton committed:
Allow accessing collection elements in format_source.

Can do it by numeric index or element name, e.g. input[0] or input["forward"]. Only works for 1-D collections.

Implements #699.

Run new and most relevant existing tests with the following commands:

```
nosetests test/unit/tools/test_actions.py
./run_tests.sh -framework -id output_format_collection
./run_tests.sh -framework -id output_format
```

jmchilton committed:
API test fix from @nsoranzo.

jmchilton committed:
Give writing job script more time to flush and sync during verification.

Still having occasional failures - https://jenkins.galaxyproject.org/job/docker-api/1172/testReport/junit/api.test_tools/ToolsTestCase/test_multirun_non_data_parameter/.

natefoo committed:
Merge branch 'release_15.10' into dev

natefoo committed:
Fix removal of comment missed in merge.

natefoo committed:
Incorporate @nsoranzo's suggestions

dannon committed:
Merge pull request #1091 from natefoo/webob-string-to-none

Fix bug with referer attribute type change in WebOb

dannon committed:
Merge pull request #1114 from jmchilton/sync_more

Give writing job script more time to flush and sync during verification.

jmchilton committed:
Merge pull request #1066 from nsoranzo/cron_ucsc

Check len files during download from UCSC.

martenson committed:
Merge pull request #1108 from yhoogstrate/cram_datatype

CRAM datatype definiton

dannon committed:
Merge pull request #1074 from martenson/lib-api-tests

[WIP] add more data libraries API tests

nsoranzo committed:
Merge branch 'master' into dev

jmchilton committed:
Don't let $input hang cheetah evaluation...

If $input is undefined. The Python input command was being used and causing Galaxy to hang during evaluation.

jmchilton committed:
Add example tools demonstrating cheetah problem (unbound variable).

Refine default override of input to raise an exception instead of just throwing in a dummy string.

jmchilton committed:
Add another cheetah syntax error example...

... ensure the test framework run the tool, detects that there is an error, and handles things appropriately.

jmchilton committed:
Revise and fix waiting for tool tests.

- maxseconds on test tag stopped being respected at some point, fix that.
- maxseconds on test tag stopped being parsed at some later point, fix that.
- Revise default behavior so it can be overridden with yet another testing environment variable GALAXY_TEST_DEFAULT_WAIT.
- Add test case to ensure there is no future regression of these behaviors.

jmchilton committed:
Merge pull request #950 from fescudie/dev

Add BIOM v1 in default datatypes.

jmchilton committed:
Merge pull request #881 from scholtalbers/role-page-fix

Fix the issue where too many datasets are linked to a certain role

jmchilton committed:
Merge pull request #1004 from jj-umn/uniprotxml_datatype

UniProtXML datatype

jmchilton committed:
Merge pull request #991 from erasche/fix-932

Fixes #932

natefoo committed:
Document GIE proxy config for nginx, mention the Ansible role.

blankenberg committed:
Merge pull request #1123 from natefoo/ie-docfix

Add nginx config for GIE proxy to documentation

Commit:
Readd jquery.migrate to galaxy.panels template because it is needed for farbtastic, which is used in visualizations.

nsoranzo committed:
Merge pull request #1119 from jmchilton/maxseconds

Revise and fix waiting for tool tests.

carlfeberhard committed:
Style: fix state-success-bg change that was not reverted during 5fd4daf

dannon committed:
Merge pull request #1129 from carlfeberhard/fix.styles

Style: fix state-success-bg change that was not reverted during 5fd4daf

dannon committed:
Rebuild style with lessc > 2.5 (which everyone needs to update to now, or we'll have battling commits of packed css

dannon committed:
Add grunt check task to verify npm modules are up to date

dannon committed:
Swap devDependencies for flat dependencies -- this allows npm ls (and thereby grunt-check-modules) to enforce dep versions.

bgruening committed:
make travis happy

hexylena committed:
Merge pull request #1131 from bgruening/IE_iobio

Add BAM iobio visualisation as Interactive Environment

dannon committed:
Merge pull request #1117 from jmchilton/input_hang

Don't let $input hang cheetah evaluation...

dannon committed:
Merge and rebuild client

dannon committed:
Fix the bad merge :(

dannon committed:
Strip whitespace

dannon committed:
Support onclick for menu items.

martenson committed:
Merge pull request #1097 from jmchilton/cwl_refactoring

Small Tool and Workflow Cleanups from CWL Work.

dannon committed:
Remove debugging logic and shortcut.

carlfeberhard committed:
UI, Lists: throttle selector animation when number of list items > 20

martenson committed:
Merge pull request #1149 from carlfeberhard/history.selector-anim

UI, Lists: throttle selector animation when number of list items > 20

carlfeberhard committed:
Merge pull request #1142 from dannon/menu_onclick_addition

Menu onclick addition

jmchilton committed:
Merge pull request #1130 from dannon/repack_dev

Enforce client build deps are up-to-date.  Rebuild style with lessc > 2.5

martenson committed:
Merge pull request #1096 from jmchilton/format_2

Outline "Format 2" workflow definitions.

martenson committed:
Merge pull request #1113 from jmchilton/format_source_collection

Allow accessing collection elements in format_source.

gregvonkuster committed:
Fix for re-installing an uninstalled TS repository with a dependency
where both the repository and the dependency were updated in the TS
after the repository was uninstalled.

martenson committed:
Merge pull request #1154 from gregvonkuster/dev

Fix for re-installing an uninstalled TS repository with a dependency

guerler committed:
Call onbeforeunload before hiding iframe through display()

Validated message

nsoranzo committed:
Fix bugs in test and code introduced in PR #1113.

jmchilton committed:
Merge pull request #1155 from nsoranzo/fix_for_1113

Fix bugs in test and code introduced in PR #1113.

martenson committed:
add local grunt-cli dependency for qunit tests

carlfeberhard committed:
Merge pull request #1159 from martenson/local-grunt-cli-dep

add local grunt-cli dependency for qunit tests

jmchilton committed:
Small optimizations to running tools.

jmchilton committed:
Save a query per output during tool execution.

jmchilton committed:
Another tool dataset creation optimization...

Rather than using three transactions to create an HDA, a dataset, and its roles - do this all in one transaction.

jmchilton committed:
Another tool execution optimization...

Prefetch all groups, roles, and associations when using User.all_users(). This reduces three queries down to one if the user has no groups or group roles and only belongs to the private user role, obviously the number of queries saved would increase more and more if the user belonged to multiple roles or groups.

blankenberg committed:
Merge pull request #1163 from jmchilton/tool_opts_1

Small optimizations to running tools.

chambm committed:
Move idpDB and MzSQlite formats from proteomics.py to binary.py, because the SQLite sniffer lives there, and more specialized formats must be registered for sniffing before the generic format
Add proper idpDB sniffing logic (akin to MzSQlite) and test case
Add H5 sniffing based on magic number and test case

chambm committed:
Move idpDB and MzSQlite formats from proteomics.py to binary.py, because the SQLite sniffer lives there, and more specialized formats must be registered for sniffing before the generic format
Add proper idpDB sniffing logic (akin to MzSQlite) and test case
Add H5 sniffing based on magic number and test case

chambm committed:
Remove dangling SQLite imports
Add h5/idpDB tests for guess_ext()

chambm committed:
Remove dangling SQLite imports
Add h5/idpDB tests for guess_ext()

chambm committed:
Update datatypes_conf.xml.sample for H5 and idpDB types
Remove extra original_filename argument in IdpDB.sniff()

chambm committed:
Update datatypes_conf.xml.sample for H5 and idpDB types
Remove extra original_filename argument in IdpDB.sniff()

nsoranzo commented on config/datatypes_conf.xml.sample:
```diff
@@ -443,8 +443,10 @@
     <sniffer type="galaxy.datatypes.binary:TwoBit"/>
     <sniffer type="galaxy.datatypes.binary:GeminiSQLite"/>
-    <sniffer type="galaxy.datatypes.proteomics:MzSQlite"/>
+    <sniffer type="galaxy.datatypes.binary:MzSQlite"/>
```
Changing the sniffer class it's not enough, you should also change the `<datatype>` element above. But probably changed all of this is not a good idea because it will break a lot of stuff for people updating.

chambm replied:
What would it break? It seems right now MzSQlite can't be sniffed at all because it's not being registered as a binary sniffer. I can roll back the MzSQLite changes if necessary though.

nsoranzo replied:
When updating to the next Galaxy release, if there is a `config/datatypes_conf.xml` file (not the sample), these changes will have to be manually merged by the admin. It's not a showstopper anyway.

chambm replied:
Wouldn't that also be true for my IdpDB change from proteomics to binary? If so, better to kill 2 birds with 1 merge, I suppose. My understanding is I that I have to move them to make the sniffer order work.

chambm committed:
Fix namespace of IdpDB and MzSQlite in datatypes_conf.xml.sample

chambm committed:
Fix namespace of IdpDB and MzSQlite in datatypes_conf.xml.sample

chambm committed:
Remove incorrect usage of subclass attribute

chambm committed:
Remove incorrect usage of subclass attribute

nsoranzo commented on config/datatypes_conf.xml.sample:
```diff
@@ -107,7 +107,7 @@
     </datatype>
     <datatype extension="toolshed.gz" type="galaxy.datatypes.binary:Binary" mimetype="multipart/x-gzip" subclass="True" />
-    <datatype extension="h5" type="galaxy.datatypes.binary:Binary" mimetype="application/octet-stream" subclass="True" display_in_upload="True"/>
+    <datatype extension="h5" type="galaxy.datatypes.binary:Binary" mimetype="application/octet-stream" display_in_upload="True"/>
```
`type` should be `galaxy.datatypes.binary:H5`

chambm replied:
That would fix the h5 sniff, but not idpDB. Why is that one still failing?

chambm committed:
Add peek functions to H5 and idpDB sniffers
Fix H5's type attribute in datatypes_conf.xml.sample

chambm committed:
Add peek functions to H5 and idpDB sniffers
Fix H5's type attribute in datatypes_conf.xml.sample

chambm commented:
Could use some help getting the tests to pass. I don't know why guess_ext is working for bam but not H5 and idpDB. As far as I can tell they're marked up the same. The only difference I can see is bam is listed under COMPRESSION_DATATYPES in sniff.py.

blankenberg committed:
Merge pull request #1166 from jmchilton/tool_opts_2

Another set of tool execution optimizations

nsoranzo commented on lib/galaxy/datatypes/binary.py:
```diff
@@ -847,8 +871,84 @@ def display_peek( self, dataset ):
+    def set_meta( self, dataset, overwrite=True, **kwd ):
+        super( IdpDB, self ).set_meta( dataset, overwrite=overwrite, **kwd )
+
+    def sniff( self, filename, original_name="" ):
```
`original_name` is not used, can be removed.

nsoranzo commented on lib/galaxy/datatypes/binary.py:
```diff
@@ -847,8 +871,84 @@ def display_peek( self, dataset ):
+    def set_meta( self, dataset, overwrite=True, **kwd ):
+        super( MzSQlite, self ).set_meta( dataset, overwrite=overwrite, **kwd )
+
+    def sniff( self, filename, original_name="" ):
```
`original_name` is not used, can be removed.

jmchilton committed:
Fixes for WorkflowContentsManager requiring app during init.

nsoranzo commented on lib/galaxy/datatypes/sniff.py:
```diff
@@ -310,6 +310,12 @@ def guess_ext( fname, sniff_order=None, is_multi_byte=False ):
     'bam'
+    >>> fname = get_test_fname('test.idpDB')
+    >>> guess_ext(fname)
+    'idpDB'
```
This should be `idpdb`.

nsoranzo commented on lib/galaxy/datatypes/binary.py:
```diff
@@ -847,8 +871,84 @@ def display_peek( self, dataset ):
+    >>> IdpDB().sniff( fname )
+    False
+    """
+    file_ext = "idpDB"
```
This should be `idpdb`.

chambm replied:
Still broken.

jmchilton committed:
Fix interface and usage of WorkflowModule.get_runtime_inputs.

Run workflow workflow form was broken for workflows with pause steps.

Fixes #1015 (thanks @afgane).

jmchilton committed:
Do not check for tool migrations when running tests.

This is an inappropriate time for the check and causes slow down when running tests.

nsoranzo committed:
Merge branch 'master' into dev

chambm committed:
Remove vestigial original_name parameters
Change idpDB extension to lower-case

chambm committed:
Remove vestigial original_name parameters
Change idpDB extension to lower-case

nsoranzo committed:
Merge branch 'master' into dev

Conflicts:
	run_tests.sh

hexylena committed:
Merge pull request #1157 from guerler/revise_iframe_unload_001

Call onbeforeunload before hiding iframe through display()

bgruening committed:
add onbeforeunload to IE

nsoranzo committed:
Merge pull request #1176 from jmchilton/auto_migrate

Do not check for tool migrations when running tests.

hexylena committed:
Merge pull request #1178 from bgruening/obbeforechange_ie

Prevent GIEs from being closed without user confirmation

souravsingh committed:
Renames reports_wsgi.ini

souravsingh committed:
Changed reports_wsgi.ini to reports.ini

souravsingh committed:
Changed reports_wsgi.ini to reports.ini

souravsingh committed:
Changed reports_wsgi.ini to reports.ini

gregvonkuster committed:
Add enhancements to the Galaxy repository install process

to cleanly handle the case described here:
https://github.com/galaxyproject/galaxy/issues/667.  Specifically, the
following scenario is now cleanly handled:

Example use case: In a toolshed I have a repository of a tool, say

package_new_gene_db_1_2_3, and in revision 1 it depends on
package_sqlite_1_0_0:

package_new_gene_db_1_2_3
package_sqlite_1_0_0

At some point in time I discover that postgres is a better solution for
my gene database, and I upload revision 2:

package_new_gene_db_1_2_3
package_postgres__2_0_0

And I remove the dependecy of sqlite.  If I go to my galaxy instance and
I remove revision 1 and install revision 2, the installation is cleanly
handled with theis PR.

It should be noted that this enhancement could adversely impact
reproducibility if best practices are not followed with regard to
defining package dependencies.  Since it is now cleanly possible to
eliminate dependencies over time, best practices must be followed to
ensure that elimination of a dependency does not affect the output of a
tool that uses the underlying hierarchy of packages.

souravsingh committed:
Updated run_reports.sh

Updated run_reports.sh to reflect new changes.

dannon committed:
Merge pull request #1126 from jgoecks/farbtastic-fix

Readd jquery.migrate to galaxy.panels template because it is needed f…

souravsingh committed:
Updated gitignore

souravsingh committed:
Updated run_reports.sh

Updated run_reports.sh to mark the latest changes.

dannon committed:
Merge pull request #1173 from jmchilton/wfcfix

Fixes for WorkflowContentsManager requiring app during init.

dannon committed:
Merge pull request #1174 from jmchilton/workflow_runtime_fix

Fix interface and usage of WorkflowModule.get_runtime_inputs.

dannon committed:
Merge pull request #1181 from souravsingh/rename_file

Renames reports_wsgi.ini

martenson committed:
Merge pull request #1193 from mvdbeek/gvk_ts_fixes

Add enhancements to the Galaxy repository install process

chambm commented:
It could also be that the customizations I've made to datatypes_conf.xml.sample are not actually being used in the test environment. Is there a way to confirm that?

natefoo committed:
Merge branch 'release_15.10' into dev

chambm committed:
Fix case in idpdb sniffer registration

chambm committed:
Merge branch 'feature/idpdb_hdf5_sniffers' of github.com:chambm/galaxy into feature/idpdb_hdf5_sniffers

# Conflicts:
#	lib/galaxy/datatypes/binary.py

chambm committed:
Add a debug print statement to see if the CI test has the same list of sniffers my local test does

blankenberg commented:
When unit tests are run. The datatypes that are loaded are those that are defined under lib/galaxy/datatypes/registry.py:Registry.set_default_values().

The sniff order is that defined under the block:

```
if len( self.sniff_order ) < 1:
            self.sniff_order = [
...
```

after which, each data type defined under:

```
       if not self.datatypes_by_extension:
            self.datatypes_by_extension = {
...
```

are added to the sniff order (in a semi-random order).

This all happens as a result of the call to sniff.py:guess_ext() that calls Registry().load_datatypes().

Its best to add these newly tested data types to both `self.sniff_order` and `self.datatypes_by_extension` in registry.py, however, just adding to only one of these places will add the sniffer to the list (but only the sniff_order list will be declarative in ordering).

chambm committed:
Add SQLite and HDF5 to the default types used when unit-testing guess_ext

chambm committed:
Revert "Add a debug print statement to see if the CI test has the same list of sniffers my local test does"

This reverts commit 2d724e843344b76d48ffd9a3d8dee0fcd3350e6d.
